src/main/kotlin/com/sourcegraph/cody/edit/FixupSession.kt

```diff
@@ -0,0 +1,354 @@
+package com.sourcegraph.cody.edit
+
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.command.undo.UndoManager
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.editor.*
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.fileEditor.FileEditor
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.util.SystemInfoRt
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.concurrency.annotations.RequiresEdt
+import com.sourcegraph.cody.agent.CodyAgent
+import com.sourcegraph.cody.agent.CodyAgentCodebase
+import com.sourcegraph.cody.agent.CodyAgentService
+import com.sourcegraph.cody.agent.CommandExecuteParams
+import com.sourcegraph.cody.agent.protocol.*
+import com.sourcegraph.cody.edit.widget.LensGroupFactory
+import com.sourcegraph.cody.edit.widget.LensWidgetGroup
+import java.util.concurrent.CancellationException
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionException
+import java.util.concurrent.TimeUnit
+
+/**
+ * Common functionality for commands that let the agent edit the code inline, such as adding a doc
+ * string, or fixing up a region according to user instructions.
+ */
+abstract class FixupSession(val controller: FixupService, val editor: Editor) : Disposable {
+  private val logger = Logger.getInstance(FixupSession::class.java)
+  val project = editor.project!!
```

Comment:
Could we please do not assume that. We had a bugs in past connected to overuse of "!!"

Reply:
Is it OK if I throw an exception there, then, if it's null?